src/android/com/adobe/phonegap/push/FCMService.java

```diff
@@ -105,6 +110,29 @@ public void onMessageReceived (RemoteMessage message) {
         PushPlugin.setApplicationIconBadgeNumber(getApplicationContext(), 0);
       }
 
+      if (toDeleteId != null) {
+        ArrayList<Map.Entry<Integer, String>> messageList = messageMap.get(Integer.parseInt(groupId));
+        if(messageList != null) {
+          NotificationCompat.Builder mBuilder = null;
+          NotificationManager mNotificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+          boolean couldRemove = messageList.removeIf(msg -> msg.getKey() == Integer.parseInt(toDeleteId));
+          if(couldRemove) {
+            if(messageList.size() == 0) {
+              String appName = getAppName(this);
+              mNotificationManager.cancel(appName, Integer.parseInt(groupId));
+                return;
+            } else {
+              extras.putBoolean(IS_DELETING, true);
+            }
+          } else {
+            return;
+          }
+        } else {
+          return;
+        }
```

Comment:
Could you please rewrite the conditionals in this piece of code so that the *uninteresting* branches come first?